src/components/Carousel.tsx

```diff
@@ -1,18 +1,92 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+};
+
+export const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [itemsOnPage, setItemsOnPage] = useState(0);
+
+  const hiddenImages = (images.length - frameSize) * itemWidth;
+
+  const getPrevImage = () => {
+    setItemsOnPage(Math.min(itemsOnPage + itemWidth * step, 0));
+
+    if (itemsOnPage === 0) {
+      setItemsOnPage(-hiddenImages);
+    }
+  };
+
+  const getNextImage = () => {
+    setItemsOnPage(Math.max(itemsOnPage - itemWidth * step, -hiddenImages));
+
+    if (itemsOnPage === -hiddenImages && infinite) {
+      setItemsOnPage(0);
+    }
+  };
+
+  const maxItemsOnPage = (images.length - frameSize) * itemWidth;
+
+  const containerWidth = frameSize * itemWidth;
+
+  return (
+    <div className="Carousel">
+      <ul
+        className="Carousel__list"
+        style={{ width: `${containerWidth}px` }}
+      >
+        {images.map(image => (
+          <li
+            className="Carousel__element"
+            key={image}
+            style={{
+              transform: `translateX(${itemsOnPage}px)`,
+              transition: `${animationDuration}ms`,
+            }}
+          >
+            <img
+              src={image}
+              alt={`${image}`}
+              style={{ width: `${itemWidth}px` }}
+            />
+          </li>
+        ))}
+
+      </ul>
+
+      <div className="Carousel__buttons">
+        <button
+          type="button"
+          className="Carousel__btn"
+          onClick={getPrevImage}
+          disabled={itemsOnPage === 0}
```

Comment:
pls see [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
          disabled={!itemsOnPage}
```